### PR TITLE
add Slepc package

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

File: var/spack/repos/builtin/packages/slepc/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    homepage = "http://www.grycap.upv.es/slepc"
+    url = "http://slepc.upv.es/download/download.php?filename=slepc-3.6.2.tar.gz"
+
+    version('3.6.2', '2ab4311bed26ccf7771818665991b2ea3a9b15f97e29fd13911ab1293e8e65df')
+
+    variant('arpack', default=False, description='Enables Arpack wrappers')
+
+    depends_on('petsc')
+    # FIXME: does not build currently, SLEPCc configure can't find Arpack's symbols: Undefined symbols for architecture x86_64: "_pdnaupd", referenced from...
+    depends_on('arpack-ng',when='+arpack') # FIXME: check if PETSc is +mpi, and if so, request arpack-ng+mpi as well.
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/slepc/package.py]: I think for the time being `FIXME` should read something like:
```
depends_on('arpack-ng~mpi',when='+arpack^petsc~mpi')
depends_on('arpack-ng+mpi',when='+arpack^petsc+mpi')
```

**Comment by davydden** [comment]
hmm, in the current version it actually hangs right after `$ spack install slepc`...

File: var/spack/repos/builtin/packages/slepc/package.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        # set SLEPC_DIR for installation
+        os.environ['SLEPC_DIR'] = self.stage.source_path
+
+        options = []
+
+        if '+arpack' in spec:
+            options.extend([
+                '--with-arpack-dir=%s' % spec['arpack-ng'].prefix.lib,
+            ])
+            if '^petsc~mpi' in spec:
```
Comment (davydden) [inline var/spack/repos/builtin/packages/slepc/package.py]: if wonder if i should do instead 
```
if `arpack-ng~mpi` in spec:`
```
?

**Comment by tgamblin** [comment]
So is this ready to go or do you want to work out the last FIXME?

**Comment by davydden** [comment]
let's merge this one https://github.com/LLNL/spack/pull/600 and I will see if I can make it work. Otherwise I would leave it as is for the moment. I never had that issue before...

**Comment by tgamblin** [comment]
Done

**Comment by davydden** [comment]
@tgamblin done. `spack install slepc+arpack ^petsc+mpi+boost ^boost+mpi` builds fine for me now.

**Pull request merged**